src/nomad_dtu_nanolab_plugin/schema_packages/sample.py

```diff
@@ -212,6 +218,41 @@ class CrystalStructure(SampleProperty):
         unit='degree',
     )
 
+    def normalize(self, archive, logger):
+        """
+        Normalizes the crystal structure by ensuring that both
+        space group number and symbol are set.
+
+        If only one of the space group number or symbol is provided, the other
+        is derived using the ASE Spacegroup class.
+
+        Parameters
+        ----------
+        archive : Archive
+            The archive object being normalized.
+        logger : Logger
+            Logger for recording normalization events or warnings.
+        """
+        super().normalize(archive, logger)
+
+        if (
+            not self.space_group_nbr > 0
+            and not self.space_group_nbr < MAX_SPACE_GROUP_NUMBER
+        ):
```

Comment:
Are you sure if the logic is correct? A simpler approach would be:

```suggestion
        if self.space_group is not None and (
            self.space_group_nbr <= 0
            or self.space_group_nbr > MAX_SPACE_GROUP_NUMBER
        ):
```